src/app/components/comparation-card/comparation-card.component.ts

```diff
@@ -0,0 +1,46 @@
+import { Component, OnInit } from '@angular/core';
+import { Actions, createEffect, ofType } from '@ngrx/effects';
+import { select, Store } from '@ngrx/store';
+import { tap } from 'rxjs/operators';
+import { Pokemon } from '../../models/pokemon.model';
+import * as PokemonActions from '../../store/actions/pokemon.actions'
+import { selectFromStore } from '../../store/selectors/pokemon.selectors';
+
+@Component({
+  selector: 'app-comparation-card',
+  templateUrl: './comparation-card.component.html',
+  styleUrls: ['./comparation-card.component.css']
+})
+export class ComparationCardComponent implements OnInit {
+  show: boolean;
+  pokemon: Pokemon;
+
+  constructor(private store: Store, private actions$: Actions) {
+    this.store.pipe(
+      select(selectFromStore)
+    ).subscribe(
+      state => { this.pokemon = state.pokemonSelected; }
+    );
+  }
+
+  ngOnInit(): void {
+  }
+
+  openCard$ = createEffect(() =>
+    this.actions$.pipe(
+      ofType(PokemonActions.selectToCompare),
+      tap(
+        () => {
+          this.show = true;
+        })
+    ), { dispatch: false }).subscribe();
```

Comment:
effects are meant to be used to dispatch other actions not to get information on when a specific action is dispatched, use the store with selectors instead